### DISCUSSION TITLE
Multiple Class of Target Classification

### DISCUSSION THREAD

**Discussion by ELITSZORD** [teamalgoritma/community#601] [Q&A] [answered]
Dear Algoritma Team, 

Di kelas C2 dimana hanya diajarkan Naive Bayes (utk target yg no multiple class / only 1&0). Dalam kasus Text Mining - Sentiment Analysis, setelah menentukan subjectivity & polarity score sebuah sentiment hasilnya kan postive/negative/neutral sentiment, jika dikasus lain kan bisa juga targetnya multiple class. Jika ingin mengklasifikasikan apakah sentiment / text tersebut masuk di kategori  postive/negative/neutral, kan bisa menggunakan Multinomial NB model bukan begitu? (atau apakah ada model lain selain Multinomial NB untuk memprediksi target Multiple Class? yg saya tau hanya LSTM)

Mohon bantuannya apakah algortima team bersedia memberikan panduan belajar step2nya untuk membuat modelnya ini di R? serta memberikan referensi2nya.  

Thank you.
Best Regards,

Faisal

**Comment by Yusraf002** [comment]
Dear @ELITSZORD ,
Terima kasih atas pertanyaannya pak, tim kami akan segera coba bantu atau carikan referensi ya pak

**Reply by IrfanChairurrachman** [reply]
Dear @ELITSZORD ,

Terima kasih atas antusiasme belajarnya. Di kelas kita mempelajari mengenai klasifikasi text seperti apakah sebuah sms masuk ke dalam spam atau tidak spam (yang terdiri dari 2 kelas). Jika kita ingin melakukan klasifikasi lebih dari 3 kelas menggunakan Naive Bayes dapat menggunakan multinomial naive bayes. Berikut referensi penerapan multinomial naive bayes di link [ini](https://rdrr.io/cran/naivebayes/man/multinomial_naive_bayes.html)

Kemudian, fungsi `NaiveBayes` dari library `e1071` yang kita gunakan di kelas juga dapat untuk melakukan klasifikasi multiclass, cara implementasinya sama seperti penerapan `NaiveBayes` yang dipelajari di kelas.

Sekian jawaban dari kami, s
emoga membantu

Best Regards,
Team Algoritma

**Reply by ELITSZORD** [reply]
Dear @IrfanChairurrachman ,

Makasih mas responsenya yg istimewa.. Disini sebenarnya yg saya maksud apakah ada model lain selain multinomial NB (naive bayes), yang untuk pengklasifikasian multiple class? Jika ada mungkin bisa di share referensi step by step pembuatan modelnya mas supaya untuk belajar temen2 disini. (mungkin contextnya masih seputar text mining / NLP ).

Kemudian pernah gk mas melebelling suatu text apakah itu sentiment negative / positve / both, kalau pernah menggunakan apa ya mas? yang saya tau cuma manual, diklasifikasikan berdasar kan kolom rating, pake textblob (tp gk bisa di R saya), dan pake VADER (tp kureng aja hasil kesimpulan sentimentnya setelah di kroscek ke textnya). 

Thank you.

Best Regards,

Faisal

**Comment by divakartika** [comment, marked as answer]
Dear @ELITSZORD ,

Multiclass classification untuk data teks dapat dilakukan dengan bermacam model, seperti Random Forest dan SVM (Support Vector Machine). Berikut kami lampirkan beberapa artikel terkait step-by-step pembuatan model baik di R maupun Python.

- R, Random Forest: https://rpubs.com/shanmukha_karthik/346007
- R, SVM: https://towardsdatascience.com/multi-class-classification-in-text-using-r-e6cf72ef1da3
- Python: https://www.analyticsvidhya.com/blog/2021/11/a-guide-to-building-an-end-to-end-multiclass-text-classification-model/

Mengenai pelabelan sentimen teks, pada R dapat dilakukan dengan fungsi `get_sentiments()` dari library `tidytext`, terdapat beberapa opsi cara pelabelan dan dapat dipilih pelabelan seperti apa yang diinginkan ([referensi](https://www.tidytextmining.com/sentiment.html)). Selain itu, kini juga ada https://datasaur.ai/, yaitu platform untuk melakukan text labelling.

Semoga jawaban ini membantu! :)

Best Regards,
Team Algoritma
